test/e2e/node_reuse_test.go

```diff
@@ -456,30 +455,6 @@ func updateNodeReuse(nodeReuse bool, m3machineTemplateName string, clusterClient
 	Expect(m3machineTemplate.Spec.NodeReuse).To(BeTrue())
 }
 
-func scaleMachineDeployment(ctx context.Context, clusterClient client.Client, oldReplicas int, newReplicas int) {
-	machineDeployments := framework.GetMachineDeploymentsByCluster(ctx, framework.GetMachineDeploymentsByClusterInput{
-		Lister:      clusterClient,
-		ClusterName: clusterName,
-		Namespace:   namespace,
-	})
-	Expect(len(machineDeployments)).To(Equal(1), "Expected exactly 1 MachineDeployment")
-	machineDeploy := machineDeployments[0]
-	Expect(*machineDeploy.Spec.Replicas).To(Equal(int32(oldReplicas)), "Expect the exact old replicas (%v) of MachineDeployment", oldReplicas)
-	patch := []byte(fmt.Sprintf(`{"spec": {"replicas": %d}}`, newReplicas))
-	err := clusterClient.Patch(ctx, machineDeploy, client.RawPatch(types.MergePatchType, patch))
-	Expect(err).To(BeNil(), "Failed to patch workers MachineDeployment")
-}
-
-func scaleControlPlane(ctx context.Context, c client.Client, name client.ObjectKey, newReplicaCount int) {
-	ctrlplane := kcp.KubeadmControlPlane{}
-	Expect(c.Get(ctx, name, &ctrlplane)).To(Succeed())
-	helper, err := patch.NewHelper(&ctrlplane, c)
-	Expect(err).To(BeNil(), "Failed to create new patch helper")
-
-	ctrlplane.Spec.Replicas = pointer.Int32Ptr(int32(newReplicaCount))
-	Expect(helper.Patch(ctx, &ctrlplane)).To(Succeed())
-}
-
 func untaintNodes(clientSet *kubernetes.Clientset, nodes *corev1.NodeList, taint *corev1.Taint) {
```

Comment:
I agree with this idea. Shoud we do that in another PR since this improvement is not necessary a part of remediation test, and this PR has already been too big now.

Reply:
Yes, that was just an idea and note for future references